### PR TITLE
Adding WDL validation with miniwdl to nightly and push tests

### PR DESCRIPTION
**Pull request by bshifaw**
miniwdl is used by wdldoc ( our soon-to-be wdl documentation tool) to validate wdl before creating an HTML doc. It's important that the wdls pass miniwdl validation so that wdldoc can proceed with documentation.

- Add wdl validation to tox tests
- Edited some the current wdls failed miniwdl validation
- Added shellcheck as one of the conda installs so that miniwdl can also validate the wdl task commands. From my testing, this shellcheck validation will not cause the miniwdl to fail but will mainly print in addition to a successfully validated WDL.

### CODE REVIEW

File: pipelines/ONT/Preprocessing/ONTFlowcell.wdl [hunk 1]
```diff
@@ -51,13 +51,11 @@ workflow ONTFlowcell {
...
     String outdir = sub(gcs_out_root_dir, "/$", "") + "/ONTFlowcell/~{dir_prefix}"
 
-    String PU = "unknown"
-    String DT = "2021-01-01T12:00:00.000000-05:00"
     if (defined(final_summary)) {
         call ONT.GetRunInfo { input: final_summary = select_first([final_summary]) }
-        String PU = GetRunInfo.run_info['instrument']
-        String DT = GetRunInfo.run_info['started']
     }
+    String PU = if defined(final_summary) then GetRunInfo.run_info['instrument'] else "unknown"
```
Comment (SHuang-Broad) [inline pipelines/ONT/Preprocessing/ONTFlowcell.wdl:57]:
```suggestion
    String PU = select_first([GetRunInfo.run_info['instrument'],  "unknown"])
```

File: pipelines/ONT/Preprocessing/ONTFlowcell.wdl [hunk 2]
```diff
@@ -51,13 +51,11 @@ workflow ONTFlowcell {
...
+    String DT = if defined(final_summary) then GetRunInfo.run_info['started'] else "2021-01-01T12:00:00.000000-05:00"
```
Comment (SHuang-Broad) [inline pipelines/ONT/Preprocessing/ONTFlowcell.wdl:58]:
```suggestion
    String DT = select_first([GetRunInfo.run_info['started'], "2021-01-01T12:00:00.000000-05:00"])
```

File: pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl [hunk 3]
```diff
@@ -8,7 +8,7 @@ import "../../../tasks/VariantCalling/GLNexus.wdl" as GLNexus
 import "../../../tasks/Utility/Hail.wdl" as Hail
 import "../../../tasks/Utility/Finalize.wdl" as FF
 
-workflow ConvertToHailMT {
+workflow RunConvertToHailMT {
```
Comment (SHuang-Broad) [inline pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl:11]: Why is this necessary? It might break things.

File: tasks/Preprocessing/Longbow.wdl [hunk 4]
```diff
@@ -238,8 +238,8 @@ task Filter {
 
         longbow filter \
             -v INFO \
-            ~{pbi_arg}~{default="" sep=" --pbi " bam_pbi} \
-            ~{model_arg}~{default="" sep=" --model " model} \
+            ~{pbi_arg}~{default="" bam_pbi} \
```
Comment (SHuang-Broad) [inline tasks/Preprocessing/Longbow.wdl:241]: Why?

File: tasks/Utility/Utils.wdl [hunk 5]
```diff
@@ -2362,13 +2362,13 @@ task FailWithWarning {
 task SplitDelimitedString {
     input {
         String s
-        String sep
+        String seperate
```
Comment (SHuang-Broad) [inline tasks/Utility/Utils.wdl:2365]: Thanks for changing this.
I believe two other workflows need to be updated accordingly.

**Review by SHuang-Broad — COMMENTED** [review]
I'm wondering why there's a need for removing `sep=...` . `miniwdl` isn't happy about it?

File: pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl (hunk 3 above)
Comment (bshifaw) [reply pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl:11]: Miniwdl complained that the task being called shouldn't be the same as the workflow name. This wdl is calling call Hail.**ConvertToHailMT** and its workflow name is **ConvertToHailMT**. I could try renaming the task being called `as` something else

File: tasks/Preprocessing/Longbow.wdl (hunk 4 above)
Comment (bshifaw) [reply tasks/Preprocessing/Longbow.wdl:241]: Miniwdl doesn't like using `sep` is normally used for Array variables on non-array variables.  Either the variables need to be changed to Array[] or `sep` needs to be removed

File: pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl (hunk 3 above)
Comment (bshifaw) [reply pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl:11]: yep, miniwdl is fine if the call name uses an alias. I'll make that change instead to avoid breaking changes

File: pipelines/ONT/Preprocessing/ONTFlowcell.wdl (hunk 1 above)
Comment (bshifaw) [reply pipelines/ONT/Preprocessing/ONTFlowcell.wdl]: womtool fails with the following error when using `select_first([])` instead of `if-else`. 
`Failed to process workflow definition 'ONTFlowcell' (reason 1 of 1): Failed to process declaration 'String PU = select_first([GetRunInfo.run_info["instrument"], "unknown"])' (reason 1 of 1): Failed to process expression 'select_first([GetRunInfo.run_info["instrument"], "unknown"])' (reason 1 of 1): Type evaluation failed for IndexAccess(IdentifierMemberAccess(GetRunInfo,run_info,Vector()),StringLiteral(instrument)). Cannot dereference a Map[String, String]? value using a String ke`y
Will need to revert it back

File: tasks/Preprocessing/CollectParentsKmerStats.wdl [hunk 6]
```diff
@@ -113,7 +113,7 @@ task ParentalReadsRepartitionAndMerylConfigure {
     }
 
     String debug_option = if (run_with_debug) then "-debug" else " "
-    String kmer_option = if (defined(kmerSize)) then ("-triobinK " + kmerSize) else " "
+    String kmer_option = if (defined(kmerSize)) then ("-triobinK " + select_first([kmerSize])) else " "
```
Comment (bshifaw) [inline tasks/Preprocessing/CollectParentsKmerStats.wdl:116]: This is due to miniwdl not liking concatenating String with any optional variable. So here we make kmerSize none-optional using select_first.

File: tasks/Utility/Utils.wdl [hunk 7]
```diff
@@ -2350,8 +2350,8 @@ task FailWithWarning {
     }
     runtime {
         cpu:                    default_attr.cpu_cores
-        memory:                 default_attr.mem_gb + " GiB"
-        disks: "local-disk " +  default_attr.disk_gb + " HDD"
+        memory:                 select_first([default_attr.mem_gb]) + " GiB"
+        disks: "local-disk " +  select_first([default_attr.disk_gb]) + " HDD"
```
Comment (bshifaw) [inline tasks/Utility/Utils.wdl:2354]: This is due to miniwdl not liking concatenating String with any optional variable. So we make the optional variable to none-optional using select_first([]). Its a little hack, happy be hear alternatives.

File: tasks/Utility/Utils.wdl (hunk 5 above)
Comment (bshifaw) [reply tasks/Utility/Utils.wdl]: done, spelling corrected to separate

File: pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl (hunk 3 above)
Comment (SHuang-Broad) [reply pipelines/TechAgnostic/Utility/ConvertToHailMT.wdl]: Thanks. I'm just afraid that someone is using the workflow and it suddenly disappear for them.

**Comment by SHuang-Broad** [comment]
Have you checked that all workflows in this branch validate with both `womtool` and `miniwdl` now?

**Comment by bshifaw** [comment]
> Have you checked that all workflows in this branch validate with both `womtool` and `miniwdl` now?

All except the deprecated_wdls folder, this folder only gets validated with womtool since I don't plan to add those wdls to the documentation site. You can see the results in the git action results [page](https://github.com/broadinstitute/long-read-pipelines/actions/runs/4546505034/jobs/8015215516), unfortunately, tox doesn't show the succeeded results unless something failed.

**Review by SHuang-Broad — APPROVED** [review]
